### PR TITLE
fix (DB): Blacksmithing Plans Loot and Conditions

### PR DESCRIPTION
**Pull request by acidmanifesto**
## Changes Proposed:
-  Per discussed no valid udb or sniff exist for loot and item.
-  Common sense approach to fill the item until further based on wowhead as research for both conditions and loot %. 
- Conditional is items will only be available if player possess blacksmithing at a skill of 285 minimal. 
- The 4 loot items with % were provided by wowhead based on time stamps of comments.

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/10384
- Closes https://github.com/chromiecraft/chromiecraft/issues/2877

## SOURCE:

Wowhead: 
https://www.wowhead.com/object=173232/blacksmithing-plans

## Tests Performed:

- Builds
- Performs

## How to Test the Changes:

1. Ensure you have blacksmith skill and skill level of 285 or higher
2. Loot Gameobject Blacksmithing Plans

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by Nyeriah — APPROVED** [review]

**Comment by acidmanifesto** [comment]
finally.
image

**Comment by IntelligentQuantum** [comment]
@acidmanifesto still empty for me :thinking:

**Comment by acidmanifesto** [comment]
> @acidmanifesto still empty for me :thinking: 

Was ur blacksmith level 285 or higher? Or lower? May have to double chdck the blacksmith id incase i got the wrong id.

**Comment by IntelligentQuantum** [comment]
> > @acidmanifesto still empty for me thinking
> 
> Was ur blacksmith level 285 or higher? Or lower? May have to double chdck the blacksmith id incase i got the wrong id.

BS 450

File: data/sql/updates/pending_db_world/rev_1643309943188748904.sql [hunk 1]
```diff
@@ -0,0 +1,74 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1643309943188748904');
+
+-- update current loot to 100% per wowhead
+UPDATE `gameobject_loot_template` SET `Chance`='100' WHERE  `Entry`=11524 AND `Item`=11614;
+UPDATE `gameobject_loot_template` SET `Chance`='100' WHERE  `Entry`=11525 AND `Item`=11615;
+UPDATE `gameobject_loot_template` SET `Chance`='100' WHERE  `Entry`=13721 AND `Item`=12827;
+UPDATE `gameobject_loot_template` SET `Chance`='100' WHERE  `Entry`=13722 AND `Item`=12830;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1643309943188748904.sql:7]: chance is not a string :D remove the single quotes

File: data/sql/updates/pending_db_world/rev_1643309943188748904.sql [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+-- delete and insert for rerun locking the loot to skill id requirement for loot
+-- none currently exist in acdb
+DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=4 AND `SourceGroup`IN ( 11524, 11525, 13721, 13722);
+INSERT INTO `conditions` SET
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1643309943188748904.sql:12]: these inserts makes no sense.

Comment (acidmanifesto) [reply data/sql/updates/pending_db_world/rev_1643309943188748904.sql:12]: right now i am waiting for more feed back on if the darn conditions are even working as intended right now.

**Review by Kitzunu — APPROVED** [review]

**Review by Nefertumm — APPROVED** [review]

**Comment by acidmanifesto** [comment]
We still need another test of this pr since the partial rewrite. if it does not work we will have to see if we have any underlying issues with the conditions as this is modeled after other conditoins dealing with profession skills being required for looting as well.

**Comment by Porchigone16** [comment]
@acidmanifesto I posted https://github.com/chromiecraft/chromiecraft/issues/2877
Im BS 300 and i have been looting this items from 3 weeks, like 30 o 40 times for day. And always the loot is empty.

**Comment by acidmanifesto** [comment]
> @acidmanifesto I posted https://github.com/chromiecraft/chromiecraft/issues/2877
> Im BS 300 and i have been looting this items from 3 weeks, like 30 o 40 times for day. And always the loot is empty.

I am aware of it. Right now i just need people to test this pr to see if it us fixed. If it doesnt then i need to know black smith level and guids of the gameobjects

**Comment by IntelligentQuantum** [comment]
@azerothcore/testers

**Comment by erikpowa** [comment]
The bug isn't that it's always empty, the problem was each blacksmith plans are spawned, while only 1 can be spawned randomly, for each instance, therefore, 3 of them should be empty.
(I have 3 plans out of 4 from BRD on ChromieCraft, I'm missing Dark Iron Mail)

**Pull request merged**